translate/storage/jsonl10n.py

```diff
@@ -416,6 +416,94 @@ def _extract_units(
             yield from parent
 
 
+class GoTextJsonUnit(BaseJsonUnit):
+    ID_FORMAT = "{}"
+
+    def getvalue(self):
+        target = self.target
+        if isinstance(target, multistring):
+            strings = list(target.strings)
+            if len(self._store.plural_tags) > len(target.strings):
+                strings += [""] * (len(self._store.plural_tags) - len(target.strings))
+            target = {
+                "select": {
+                    "feature": "plural",
+                    "arg": "N",
+                    "cases": {
+                        plural: {"msg": strings[offset]}
+                        for offset, plural in enumerate(self._store.plural_tags)
+                    },
+                }
+            }
+        value = {"id": self.getid()}
+        if self.notes:
+            value["translatorComment"] = self.notes
+        value["translation"] = target
+        return value
+
+
+class GoTextJsonFile(JsonFile):
+    """gotext JSON file
+
+    See following URLs for doc:
+
+    https://pkg.go.dev/golang.org/x/text/cmd/gotext
+    https://github.com/golang/text/tree/master/cmd/gotext/examples/extract/locales/en-US
+    """
+
+    UnitClass = GoTextJsonUnit
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        locale = self.gettargetlanguage()
```

Comment:
You should implement gettargetlanguage/settargetlanguage to operate on `data['language']`.

Reply:
Working on this now; thank you!

Reply:
I pushed a possible fix; it doesn't work in the way you suggested, but LMK what you think. When trying to override gettargetlanguage I couldn't figure out if we had the whole file parsed already or not, so I thought it might make sense to allow it to be file based (it's normally stored in language name folders anyways, so this will work), but then override it if we detect a language in the file when we parse.